Kernel/Arch/aarch64/Processor.cpp

```diff
@@ -255,11 +258,31 @@ void Processor::switch_context(Thread*& from_thread, Thread*& to_thread)
     dbgln_if(CONTEXT_SWITCH_DEBUG, "switch_context <-- from {} {} to {} {}", VirtualAddress(from_thread), *from_thread, VirtualAddress(to_thread), *to_thread);
 }
 
+extern "C" FlatPtr do_init_context(Thread* thread, u32 new_interrupts_state)
+{
+    VERIFY_INTERRUPTS_DISABLED();
+
+    Aarch64::SPSR_EL1 spsr_el1 = {};
+    memcpy(&spsr_el1, (u8 const*)&thread->regs().spsr_el1, sizeof(u64));
```

Comment:
Why do we have to do this using memcpy instead of defining assignment operators on these objects?

Reply:
No real reason I guess, it seemed pretty akward to me as well, but that's how it was done elsewhere so far.

Reply:
Also why is it using `sizeof(u64)` instead of `sizeof(Aarch64::SPSR_EL1)`, they might be the same but just `u64` seems arbitrary,
Same 2 lines down

Reply:
`ThreadRegisters::spsr_el1` is `u64`, so it's not arbitrary, but I agree, it's ugly.
I just don't want to go and introduce the operators and modify all usages in the Kernel in this PR.